frontend/src/components/housing/submission/SubmissionIntakeForm.vue

```diff
@@ -201,8 +208,18 @@ async function onSaveDraft(data: any, isAutoSave = false) {
   editable.value = false;
 
   const tempData = Object.assign({}, data);
+
+  // Cleanup unneeded data to be saved to draft
   delete tempData['addressSearch'];
 
+  // Remove empty permits
+  if (Array.isArray(tempData.appliedPermits)) {
+    tempData.appliedPermits = tempData.appliedPermits.filter((x: Partial<Permit>) => x?.permitTypeId);
+  }
+  if (Array.isArray(tempData.investigatePermits)) {
+    tempData.investigatePermits = tempData.investigatePermits.filter((x: Partial<Permit>) => x?.permitTypeId);
+  }
+
```

Comment:
Theres a front end util function called `omit` which hides the delete ugliness. This could be rewritten as 
```ts
const tempData = omit(data, ['addressSearch']);
```

Also `tempData` should be renamed. This is the data that is actually being submitted.